docs/custom-filters-v1.md

```diff
@@ -0,0 +1,47 @@
+# Custom filters V1
+
+Custom filters are Javascript code snipets that will be embedded into gitStream's functionality after a review and approval proccess by our engineers
+
+!!! attention
+
+    :octicons-beaker-24: Coming soon
+
+
+###  Creating filters
+
+Filters can have at most one input parameter such as the available [context variables](context-variables.md), that is provided to the filter in the `.cm` file using a pipe.
+
+Filters must return a valid Javascript type (i.e Bool, Int, String, Object, etc...)
+
+In addition to the code implemetion of the filter, the filter creator must specify the npm dependecies in JSON format, as it appears in the `package_lock.json` file
```

Comment:
This is something TBD, as we might want to limit imports to built-in imports etc.

Reply:
can't we add the other dependencies to our package_lock.json?

Reply:
there might be few implications, this is why I suggest to have TBD